### PR TITLE
Add migration to add primary parameter to update managed targets moun…

### LINKED ISSUE #418: Upgrade issue from 3.1 to 4.0
In the patch that added MMP support to ZFS (4.0), we added a new field to the kwargs that are passed in with `UpdateManagedTargetMount`. 

The class (and kwargs) are ultimately pickled and saved in the db as a text field under `chroma_core_step_result.args`.

This causes an issue after upgrading to 4.0 as when the (3.1) field is hydrated back to a dict it's missing the `primary` key and throws.

We need to figure out a way of either working around that field, or deriving it's value from 3.1 data.

#419  fixes this

### PR DESCRIPTION
**Pull request by tanabarr**
…t to fix update tests

fixes #418 

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

---
This change is [Reviewable](https://reviewable.io/reviews/intel-hpdd/intel-manager-for-lustre/419)

### CODE REVIEW

File: chroma-manager/chroma_core/migrations/0038_auto__add_primary_umtm_steps.py [hunk 1]
```diff
@@ -0,0 +1,1503 @@
...
+from toolz.curried import filter as cfilter
+from toolz.curried import map as cmap
+
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        # get stepresults for UpdateManagedTargetMount class and update stepresults to include 'primary' arg
+        pipe(orm['chroma_core.stepresult'].objects.all(),
+             cfilter(lambda x: 'UpdateManagedTargetMount' in str(x.step_class)),
```
Comment (johnsonw) [inline chroma-manager/chroma_core/migrations/0038_auto__add_primary_umtm_steps.py]: Nice

File: chroma-manager/chroma_core/migrations/0038_auto__add_primary_umtm_steps.py [hunk 2]
```diff
@@ -0,0 +1,1503 @@
+# -*- coding: utf-8 -*-
+import cPickle as pickle
```
Comment (jgrund) [inline chroma-manager/chroma_core/migrations/0038_auto__add_primary_umtm_steps.py]: Is this import needed?

Comment (tanabarr) [reply chroma-manager/chroma_core/migrations/0038_auto__add_primary_umtm_steps.py]: it's a better performing implementation and I have been in the habit of using it in preference to the original implementation

Comment (jgrund) [reply chroma-manager/chroma_core/migrations/0038_auto__add_primary_umtm_steps.py]: Sure, I just meant I don't see `pickle` being used directly anywhere.

File: chroma-manager/tests/integration/installation_and_upgrade/test_data_consistancy.py [hunk 3]
```diff
@@ -10,6 +11,7 @@ def test_access_all_data_all_endpoints(self):
                                 '/api/test_host/']
 
         end_points = self.get_json_by_uri("/api/", args={'limit': 0})
+        time.sleep(7200)
```
Comment (jgrund) [inline chroma-manager/tests/integration/installation_and_upgrade/test_data_consistancy.py]: Do we need this sleep here?

File: chroma-manager/chroma_core/migrations/0039_add_primary_umtm_stepresults.py [hunk 4]
```diff
@@ -0,0 +1,1505 @@
+# -*- coding: utf-8 -*-
+import cPickle as pickle
```
Comment (jgrund) [inline chroma-manager/chroma_core/migrations/0039_add_primary_umtm_stepresults.py]: Sorry to keep harping on this, but is this import being used anywhere?

Comment (tanabarr) [reply chroma-manager/chroma_core/migrations/0039_add_primary_umtm_stepresults.py]: it is a quicker implementation than the standard pickle module, I'm in the habit of using it

File: chroma-manager/chroma_core/migrations/0039_add_primary_umtm_stepresults.py [hunk 5]
```diff
@@ -0,0 +1,1499 @@
+# -*- coding: utf-8 -*-
+from south.v2 import DataMigration
+from toolz import pipe
+from toolz.curried import filter as cfilter
```
Comment (jgrund) [inline chroma-manager/chroma_core/migrations/0039_add_primary_umtm_stepresults.py]: We don't need toolz here anymore.

**Pull request merged**